armi/reactor/converters/blockConverters.py

```diff
@@ -497,8 +510,8 @@ def convert(self):
             )
         )
         self._dissolveComponents()
-        numRings = locations.minimumRings(
-            self._sourceBlock, self._sourceBlock.getNumPins()
+        numRings = self._sourceBlock.spatialGrid.getMinimumRings(
```

Comment:
We cannot assume that a block has a spatialGrid in the first place (you set up the test blocks below so that they do, but in general, Blocks dont need to have one). Furthermore, there isn't really any guarantee if a Block _does_ have one, that it will be a type of grid that corresponds in any way to the type of the block. Hex blocks with some sort of ThR grid is something that I've seen in the wild before.

That's not to say the the existing implementation here is handling any of this nuance properly anyhow; it's just stuffing all of those assumptions into the remnants of locations.py and holding on for dear life. I just see reaching for a spatial grid that might not be there, and which this converter is really expecting to correspond to the geometry of the block itself opens us up to lots of new surprises/bugs down the line.

I think that maybe we should just be more honest here in the block converters about their assumptions and limitations and confirm that the input blocks satisfy those assumptions. That might look something like:
 - move the type-to-geom mapping logic out of locations.py and into here; I think this is the only place that's really using it anyway so might as well own it.
 - check if things do have a spatialGrid, and if so, make sure that it matches the geom type of the block. otherwise these converters will be doing something wrong.
 - for the cartesian case (do we really exercise this??), perhaps _require_ that the block has a spatialGrid, so that we can stop assuming that `throughCenter == True`.